src/main/java/bio/terra/service/snapshot/flight/create/CreateSamGroupStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+
+public class CreateSamGroupStep implements Step {
+  private final IamService iamClient;
```

Comment:
nit: Can we name this iamService? Probably just me, but the name of this made me think it was directly making API call instead of using the service methods.
```suggestion
  private final IamService iamService;
```